tst/com/amazon/corretto/crypto/provider/test/EdDSATest.java

```diff
@@ -117,81 +201,130 @@ public void keyFactoryValidation() throws GeneralSecurityException {
-    byte[] message, signatureACCP, signatureJCE;
-    Random random = new Random();
-
-    for (int messageLength = 1; messageLength <= 1024; messageLength++) {
-      message = new byte[messageLength];
-      random.nextBytes(message);
-      // Sign with ACCP and verify with SunEC
-      nativeSig.initSign(privateKey);
-      nativeSig.update(message, 0, message.length);
-      signatureACCP = nativeSig.sign();
-      jceSig.initVerify(publicKey);
-      jceSig.update(message);
-      assertTrue(
-          jceSig.verify(signatureACCP),
-          "Native->JCE: Ed25519 (message length: " + messageLength + ")");
-
-      // Sign with SunEC and verify with ACCP
-      jceSig.initSign(privateKey);
-      jceSig.update(message, 0, message.length);
-      signatureJCE = jceSig.sign();
-      nativeSig.initVerify(publicKey);
-      nativeSig.update(message);
-      assertTrue(
-          nativeSig.verify(signatureJCE),
-          "JCE->Native: Ed25519 (message length: " + messageLength + ")");
-
-      assertArrayEquals(signatureJCE, signatureACCP);
-    }
+    testInteropValidation(nativeSig, jceSig, false);
   }
 
   @Test
   public void bcInteropValidation() throws GeneralSecurityException {
-    // Generate keys with ACCP and use BC KeyFactory to get equivalent Keys
     final Signature nativeSig = Signature.getInstance("Ed25519", NATIVE_PROVIDER);
     final Signature bcSig = Signature.getInstance("Ed25519", BOUNCYCASTLE_PROVIDER);
+    testInteropValidation(nativeSig, bcSig, false);
+  }
+
+  @Test
+  public void selfValidationPh() throws GeneralSecurityException {
+    assumeTrue(ed25519phIsEnabled());
+    final Signature nativeSignerSig = Signature.getInstance("Ed25519ph", NATIVE_PROVIDER);
+    final Signature nativeVerifierSig = Signature.getInstance("Ed25519ph", NATIVE_PROVIDER);
+    testInteropValidation(nativeSignerSig, nativeVerifierSig, true);
+  }
+
+  @Test
+  public void jceInteropValidationPh() throws GeneralSecurityException {
+    assumeTrue(ed25519phIsEnabled());
+    final Signature nativeSig = Signature.getInstance("Ed25519ph", NATIVE_PROVIDER);
+    final Signature jceSig = Signature.getInstance("Ed25519", "SunEC");
+    makeJceSignaturePh(jceSig);
+    testInteropValidation(nativeSig, jceSig, true);
+  }
+
+  @Test
+  public void bcInteropValidationPh() throws GeneralSecurityException {
+    assumeTrue(ed25519phIsEnabled());
+    final Signature nativeSig = Signature.getInstance("Ed25519ph", NATIVE_PROVIDER);
+    testInteropValidation(nativeSig, bcPrehashSig, true);
+  }
+
+  @Test // sanity check to assert that JCE and BC are interoperable
+  public void bcJceInteropValidationPh() throws GeneralSecurityException {
+    assumeTrue(ed25519phIsEnabled());
+    final Signature jceSig = Signature.getInstance("Ed25519", "SunEC");
+    makeJceSignaturePh(jceSig);
+    testInteropValidation(jceSig, bcPrehashSig, true);
+  }
+
+  public void testInteropValidation(Signature signer, Signature verifier, boolean preHash)
```

Comment:
Nit: This test tests both directions so there's no distinction between signer/verifier in the interface.

Reply:
that's a fair point. update.